index.js

```diff
@@ -88,11 +89,32 @@ const claimFunds = async (self, amount, signature) => {
         .removeListener('transaction', handleTransaction))
       resolve()
     }
-
     self.api.connection.on('transaction', handleTransaction)
   })
 }
 
+async function reloadIncomingChannelDetails (ctx) {
```

Comment:
Shouldn't this also check things like the channel expiry and balance?

Reply:
I added a check that the balance is what we expect. With regard to the channel expiry, I check that the settleDelay is high enough so that the plugin has enough time to submit its best claim.

Reply:
I'm actually not sure that's good enough. Take a look at the [paychan tutorial](https://ripple.com/build/payment-channels-tutorial/#2-the-payee-checks-specifics-of-the-payment-channel). If the `expiration` or `cancelAfter` fields are set, that means the channel already has an expiry. The purpose of the `settleDelay` is just to make sure that _when_ someone sets the expiration it's at least that far in the future from that point. But that point could already have passed when the plugin is checking it.

Reply:
Added some more validation. The channel is now validated as recommended in the paychan tutorial.